pkg/audit/audit.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/vivint/infectious"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/dht"
+	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/piecestore/rpc/client"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/transport"
+)
+
+var mon = monkit.Package()
+
+// Share is an erasure share
+type Share struct {
+	Error       error
+	PieceNumber int
+	Data        []byte
+}
+
+// Auditor implements the downloader interface
+type Auditor struct {
+	downloader Downloader
+}
+
+// Downloader enables downloading shares
+type Downloader interface {
+	DownloadShares(ctx context.Context, pointer *pb.Pointer, stripeIndex int, nodes []*pb.Node) (shares []Share, err error)
+	lookupNodes(ctx context.Context, pieces []*pb.RemotePiece) (nodes []*pb.Node, err error)
```

Comment:
you shouldn't have a `lookupNodes` on it, let the `DownloadShares` method handle it internally

Reply:
okay so put the code from inside lookupNodes into the DownloadShares function?

Reply:
yes

Reply:
<img width="717" alt="screen shot 2018-10-08 at 10 21 02 am" src="https://user-images.githubusercontent.com/26885324/46614534-134d7e00-cae4-11e8-9c0c-abf652cae4b7.png">
I'm trying this but then I have no access to the overlay client because this function is a method on Auditor.